gateway-spi/src/main/java/org/apache/knox/gateway/dispatch/DefaultHttpClientFactory.java

```diff
@@ -119,6 +126,19 @@ public HttpClient createHttpClient(FilterConfig filterConfig) {
     // See KNOX-1530 for details
     builder.disableContentCompression();
 
+    if (filterConfig.getInitParameter(PARAMETER_RETRY_COUNT) != null
+        && StringUtils
+        .isNumeric(filterConfig.getInitParameter(PARAMETER_RETRY_COUNT))) {
```

Comment:
nit: this logic worths a private boolean method IMO

Reply:
yeah, I can brach this out to a private method.